snuba/web/db_query.py

```diff
@@ -388,6 +388,48 @@ def get_query_cache_key(formatted_query: FormattedQuery) -> str:
     return md5(force_bytes(formatted_query.get_sql())).hexdigest()
 
 
+def check_sorted_sql_key_in_cache(
+    formatted_query_sorted: Optional[FormattedQuery], reader: Reader
+) -> bool:
+    cache_partition = _get_cache_partition(reader)
+    if formatted_query_sorted:
+        key_sorted = get_query_cache_key(formatted_query_sorted)
+        percentage = state.get_config(
+            "sort_sql_fields_and_conditions_rollout_percentage", 0
+        )
+        assert isinstance(percentage, int)
+        if hash_to_probability(key_sorted) < percentage:
+            key_sorted_with_prefix = f"sorted:{key_sorted}"
+            result = cache_partition.get(key_sorted_with_prefix)
+            if result is not None:
+                return True
+            else:
+                empty_result: Result = {
+                    "meta": [],
+                    "data": [],
+                    "totals": {},
+                    "profile": None,
+                    "trace_output": "",
+                }
+                cache_partition.set(key_sorted_with_prefix, empty_result)
+                return False
+    return False
+
+
+def hash_to_probability(hexadecimal: str) -> float:
+    MAX_HASH_PLUS_ONE = 16 ** len(hexadecimal)
```

Comment:
To get the max float value of the MD5 hash, we convert the 32-character hexadecimal to decimal by raising 16 to power of 32 (length of MD5). Since the MD5 hash returns a uniformly distributed number from 0 to the max float value, we just convert the hexadecimal to a float and divide by the max value to get a random decimal from [0,1).

My initial thought was to calculate it each time in the case we switch hashing algorithms (e.g. SHA-256) in the future, but I guess that will be unlikely. Updated as global.